src/libutil/experimental-features.cc

```diff
@@ -221,6 +221,21 @@ constexpr std::array<ExperimentalFeatureDetails, 14> xpFeatureDetails = {{
             Allow parsing of timestamps in builtins.fromTOML.
         )",
     },
+    {
+        .tag = Xp::ReadOnlyLocalStore,
+        .name = "read-only-local-store",
+        .description = R"(
+            Allow the use of the `read-only` parameter in local store URIs.
+
+            Set this parameter to `true` to allow stores with databases on read-only filesystems to be opened for querying; ordinarily Nix will refuse to do this.
+
+            This is because SQLite requires write access to the database file to perform the file locking operations necessary for safe concurrent access.
+            When `read-only` is set to `true`, the database will be opened in immutable mode.
+
+            Under this mode, SQLite will not do any locking at all, so you should be certain that the database will not be changed.
+            While the filesystem the database resides on might be read-only to this process, consider whether another user, process, or system, might have write access to it.
```

Comment:
```suggestion
```

None of this needs to be documented in the experimental feature flag, but should instead be local to the actual setting. Also currently it's not only slightly confusing, because those sentences say almost (exactly?) the same thing but in very different words. Which version is authoritative?

Reply:
The setting documentation was closer to authoritative so I've gone with that version (but with an extra sentence from the flag documentation that I think is useful).